### PR TITLE
Storybook Tool

### PR DESCRIPTION
**Pull request by puntope**
### Changes proposed in this Pull Request:

Initial PR adding Project Templates as well as Storybook implementation

### Detailed test instructions:

1.  Uninstall all the storybook dependencies in GLA
2.  Install this branch in the project using `npm i -D  'https://gitpkg.now.sh/woocommerce/grow/storybook?add/storybook'`  
3. Check in `node_modules` you have `woocommerce-grow-story-book` folder
4.  Check in `node_modules/bin` you have `storybook.js` 
5. Replace storybook lines in scripts with this in GLA package JSON
```
"storybook": "storybook",
"storybook:build": "storybook build",
"storybook:deploy": "storybook deploy",
```
6. Execute `npm tun storybook` and see how the storybook watch initialises and you can access the URL
7. Execute `npm tun storybook:build` and see how the storybook generates dist directory
8. Execute `npm tun storybook:deploy` and see how the storybook deploys the dist version to GH pages in https://woocommerce.github.io/google-listings-and-ads/?path=/story/tree-select-control--base 
9. Check the templates generated with `yo:ventures`

### Changelog entry

> Add - Storybook
> Add - Repository Templates

### CODE REVIEW

**Comment by tomalec** [comment]
:nail_care: I'm currently baking [a PR](https://github.com/woocommerce/grow/pull/2) to add [Yeoman generator](https://github.com/tomalec/generator-ventures) here would you mind skipping the first commit https://github.com/woocommerce/grow/pull/1/commits/50f7260d23a4f843585be17d0aa9dc94da8a377c
So adding a generator will use the local/updated generator? 
It's not a big deal, I can update it after this one, this is just my OCD of keeping history cleaner, as I'd change the dotfiles content to use the project name to
:construction: "~Ventures~ Grow Packages"

:question:  Also I was thinking of adding individual packages to `/packages/js/storybook` `/packages/php/something_for_composer_to_use` to follow woocommerce/woocommerce example https://github.com/woocommerce/woocommerce/tree/trunk/packages, WDYT?

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+lts/fermium
```
Comment (tomalec) [inline .nvmrc]: :question: Does the storybook strictly requires Node 14? It's in maintenance mode. The latest active LTS is 16, and the current is 18 https://nodejs.org/en/about/releases/

File: storybook/utils/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const shell = require("shelljs");
+const path = require("path")
+
+process.env.PATH += ( path.delimiter + path.join( process.cwd(), 'node_modules', '.bin') );
+
+module.exports = {
+	shell
+}
```
Comment (tomalec) [inline storybook/utils/index.js]: WDYT of renaming it to `/storybook/utils.js` to avoid having too many `index.js`es everywhere?

**Comment by tomalec** [comment]
:nail_care: Have you considered ES Modules instead of CommonJS syntax? https://github.com/woocommerce/grow/compare/add/storybook...add/storybook-esm?expand=1
So eventually some day, we could forget about legacy stuff and heavy transpiling ;)

**Review by tomalec — COMMENTED** [review]
Tested locally. Works nice. I left few cosmetic :nail_care: comments. The only blocking :construction:  one is about "~Ventures~ Grow Packages" name, plus a few questions :question: .

:nail_care: Also, using this package as a dependency removes coloring from the console :(  
is there something easy we can do about it?
image

**Comment by tomalec** [comment]
I did a quick check and the `shelljs-live` proposed in https://github.com/shelljs/shelljs/issues/86#issuecomment-939378750 seems fine. The question is would be like to rely on such unpopular package.

File: .nvmrc (hunk 1 above)
Comment (puntope) [reply .nvmrc]: Updated to Gallium (v16)

File: storybook/utils/index.js (hunk 2 above)
Comment (puntope) [reply storybook/utils/index.js]: >  about legacy stuff and heavy transpiling ;)

What do you mean, do you have some examples about this? 

I changed it tho in last commits in advance.

**Comment by puntope** [comment]
>  about legacy stuff and heavy transpiling ;)

What do you mean, do you have some examples about this? 

I changed it tho in last commits in advance.

**Comment by puntope** [comment]
> 💅 I'm currently baking https://github.com/woocommerce/grow/pull/2 to add [Yeoman generator](https://github.com/tomalec/generator-ventures) here would you mind skipping the first commit https://github.com/woocommerce/grow/commit/50f7260d23a4f843585be17d0aa9dc94da8a377c

No problem. I removed the files referent to that commit.

**Comment by puntope** [comment]
> 💅 Also, using this package as a dependency removes coloring from the console :(
is there something easy we can do about it?

> I did a quick check and the shelljs-live proposed in https://github.com/shelljs/shelljs/issues/86#issuecomment-939378750 seems fine. The question is would be like to rely on such unpopular package.

For now i'd prefer to keep it without fancy colors. Since my goal is to have this done in current cooldown, getting rid of Storybook deps in GLA.  But definitely I'm down for experimenting with some tools to improve the experience in a future PR.

**Comment by puntope** [comment]
>  Also I was thinking of adding individual packages to /packages/js/storybook /packages/php/something_for_composer_to_use to follow woocommerce/woocommerce example https://github.com/woocommerce/woocommerce/tree/trunk/packages, WDYT?

Sure no problem, I changed that in last commits. 👌

**Comment by puntope** [comment]
Howdy @tomalec  Thanks for the suggestions.

This is ready for another review 👌

File: packages/js/storybook/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Grow Team Storybook
+
+Storybook dependencies and basic scripts
+
+
+## Prerequisites
+
+- You will the StoryBook config files as if you installed it in your ouw repository.
+- You will need to configure GH Pages s if you would do it in your own repository.
+- You will need to add the scripts in your repo calling the storybook binaries described in the usage section.
```
Comment (tomalec) [inline packages/js/storybook/README.md]: :construction: 
I'd remove repeated, "You will". Also, AFAIK  "Prerequisites" means that something needs to be done as a prior condition.

Also, I'd remove the last point, as it is more of a "usage" than a prerequisite.
```suggestion
- Create StoryBook config files as if you installed it in your own repository.
- Configure GH Pages as if you would do it in your own repository.
```

File: packages/js/storybook/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+	"name": "woocommerce-grow-story-book",
+	"title": "WooCommerce Grow - Storybook Package",
+  	"type": "module",
+	"license": "GPL-3.0-or-later",
+	"version": "1.12.5",
```
Comment (tomalec) [inline packages/js/storybook/package.json]: I think we should start versioning from scratch. I guess it's just leftover from GLA, isn't it?
```suggestion
	"version": "0.0.1",
```

File: packages/js/storybook/bin/storybook.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * External dependencies
+ */
+import process from 'process';
+
+/**
+ * Internal dependencies
+ */
+import { shell } from '../utils.js'
+
+const argv = process.argv;
```
Comment (tomalec) [inline packages/js/storybook/bin/storybook.js]: :nail_care: 
```suggestion
import { argv } from 'process';

/**
 * Internal dependencies
 */
import { shell } from '../utils.js'
```

File: packages/js/storybook/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+	"name": "woocommerce-grow-story-book",
```
Comment (tomalec) [inline packages/js/storybook/package.json]: I'd change it to `-storybook` as the original package name
```suggestion
	"name": "woocommerce-grow-storybook",
```

**Review by tomalec — APPROVED** [review]
Tested locally, reviewed the code LGTM

**Pull request merged**